packages/web/src/js/Tooltip.ts

```diff
@@ -322,9 +353,79 @@ class Tooltip extends BaseComponent {
       });
     }
   }
+
+  autoCloseHandler = (event: Event) => {
+    const id = this.tip.getAttribute('id') as string;
+    const trigger = document.querySelector(`[data-spirit-target="#${id}"]`) as HTMLElement;
+
+    const shouldClose = trigger && clickOutsideElement(trigger, event);
+    if (event.target && shouldClose) {
+      this.activeTrigger[TRIGGER_CLICK] = false;
+      this.activeTrigger[TRIGGER_HOVER] = false;
+      this.leave();
+    }
+  };
+
+  isWithActiveTrigger() {
+    return Object.values(this.activeTrigger).includes(true);
+  }
+
+  enter() {
+    if (this.isShown() || this.isHovered) {
+      this.isHovered = true;
+
+      return;
+    }
+
+    this.isHovered = true;
+
+    this.show();
+  }
+
+  leave() {
+    if (this.isWithActiveTrigger()) {
+      return;
+    }
+
+    this.isHovered = false;
+
+    this.hide();
+  }
+
+  addEventListeners() {
+    const id = this.tip.getAttribute('id') as string;
+    const trigger = document.querySelector(`[data-spirit-target="#${id}"]`) as HTMLElement;
+    const { enableHover } = this.config as Config;
+
+    EventHandler.on(document, EVENT_CLICK, (event: Event) => this.autoCloseHandler(event));
+
+    EventHandler.on(trigger, EVENT_CLICK, () => {
+      const context = Tooltip.getOrCreateInstance(this.element as HTMLElement);
+      context.toggle();
+    });
+
+    if (enableHover) {
+      this.addMouseEventListeners();
+    }
+  }
+
+  addMouseEventListeners() {
+    const id = this.tip.getAttribute('id') as string;
+    const trigger = document.querySelector(`[data-spirit-target="#${id}"]`) as HTMLElement;
```

Comment:
Hmm, again. Definitely, we need a function.

Reply:
function added